code/modules/projectiles/gun_system.dm

```diff
@@ -537,6 +535,8 @@ User can be passed as null, (a gun reloading itself for instance), so we need to
 
 ///Set the target and take care of hard delete
 /obj/item/weapon/gun/proc/set_target(atom/object)
+	if(gun_user.Adjacent(object))
```

Comment:
it will cancel it if they push foward and it doesn't update however ,i  think if you can't do better than this then to not change it..

Reply:
it always update? i